ansible_base/models/resource.py

```diff
@@ -0,0 +1,122 @@
+import uuid
+
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.db.models.functions import Concat
+from django.db.utils import ProgrammingError
+
+from .service_id import service_id
+
+_resource_registry = None
+
+
+# Circular imports make this really hard to initialize correctly.
+# I don't know why, but I can't put this function anywhere else without inducing
+# circular imports.
+def get_registry():
+    global _resource_registry
+    if _resource_registry is None:
+        from django.conf import settings
+
+        if hasattr(settings, "RESOURCE_REGISTRY_CONFIG"):
```

Comment:
Should we prefix this setting with ANSIBLE_BASE? There are a couple of "settings" without the prefix but these were mostly copied from AWX.